main.tf

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  jenkins_metadata = {
+    bitnami-base-password  = "${local.jenkins_password}"
+    status-uptime-deadline = 420
+    startup-script         = "${data.template_file.jenkins_startup_script.rendered}"
+  }
+
+  jenkins_password                = "${coalesce(var.jenkins_initial_password, random_string.jenkins_password.result)}"
+  jenkins_startup_script_template = "${file("${path.module}/templates/jenkins_startup_script.sh.tpl")}"
+  jenkins_tags                    = ["${var.jenkins_instance_network_tag}"]
+  jenkins_username                = "user"
+
+  jenkins_workers_agent_attach_script = <<EOF
```

Comment:
Let's pull the worker startup script into its own template file instead of inlining.